src/tls.c

```diff
@@ -7867,7 +7868,10 @@ static word16 TLSX_PreSharedKey_Write(PreSharedKey* list, byte* output,
          * The binders are based on the hash of all the ClientHello data up to
          * and include the identities written above.
          */
-        idx += TLSX_PreSharedKey_GetSizeBinders(list, msgType);
+        len = ret = TLSX_PreSharedKey_GetSizeBinders(list, msgType);
```

Comment:
The `TLSX_PreSharedKey_GetSizeBinders` function return type is `word16`. I have a feeling the `return (word16)SANITY_MSG_E;` is going to result in a positive `len`?

Would it be better to have the `TLSX_PreSharedKey_GetSizeBinders`, `TLSX_PreSharedKey_WriteBinders`, `TLSX_PskKeModes_GetSize`, `TLSX_PskKeModes_Write` functions return `int` to avoid any confusion?

Also there are still functions in tls.c that have this issue... `TLSX_PostHandAuth_GetSize`, `TLSX_PostHandAuth_Write`, `TLSX_PostHandAuth_Parse`, `TLSX_EarlyData_GetSize`, `TLSX_EarlyData_Write`, `TLSX_EarlyData_Parse`.

Reply:
Changed functions so that length/offset is a parameter and return is only error code.